### PR TITLE
remove duplicate code

### PR DESCRIPTION
**Pull request by CaoDonghui123**
#### What type of PR is this?

/kind cleanup
#### What this PR does / why we need it:
remove duplicate code 

https://github.com/kubernetes/kubernetes/blob/513ae557a3231cde76ad155bbbd4e18f8182df25/staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go#L148

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Review by pacoxu — CHANGES_REQUESTED** [review]
See https://github.com/kubernetes/kubernetes/pull/16160#discussion_r42894642. 

 #16160 seems to be the reason why we close it here.
/hold

**Comment by CaoDonghui123** [comment]
>// This file descriptor needs to close so the next process (Launch) can claim it.
in my opinion, this is not a necessary reason

File: staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go [hunk 1]
```diff
@@ -151,8 +151,6 @@ func (e Editor) LaunchTempFile(prefix, suffix string, r io.Reader) ([]byte, stri
 		os.Remove(path)
 		return nil, path, err
 	}
-	// This file descriptor needs to close so the next process (Launch) can claim it.
-	f.Close()
```
Comment (wzshiming) [inline staging/src/k8s.io/kubectl/pkg/cmd/util/editor/editor.go:155]: > This file descriptor needs to close so the next process (Launch) can claim it.

I think this is not a duplicate

**Review by wzshiming — CHANGES_REQUESTED** [review]

**Comment by brianpursley** [comment]
IMO, its confusing because `LaunchTempFile` is doing two different things (creating the temp file from the stream, and launching) and the defer boundary isn't helpful for releasing the lock.

What if the code is refactored into something like this so that `defer f.close()` works instead of having to call f.close() inline?

Something like this (haven't tested it)...

```
func (e Editor) LaunchTempFile(prefix, suffix string, r io.Reader) ([]byte, string, error) {
	var path string
	var err error
	if path, err = createTempFileFromStream(prefix+"*"+suffix, r); err != nil {
		return nil, "", err
	}
	if err := e.Launch(path); err != nil {
		return nil, path, err
	}
	bytes, err := ioutil.ReadFile(path)
	return bytes, path, err
}

// This is refactored out of LaunchTempFile so that the defer is executed at the right time to release the file lock after the temp file is created
func createTempFileFromStream(pattern string, r io.Reader) (string, error) {
	f, err := os.CreateTemp("", pattern)
	if err != nil {
		return "", err
	}
	defer f.Close()
	path := f.Name()
	if _, err := io.Copy(f, r); err != nil {
		os.Remove(path)
		return "", err
	}
	return path, nil
}
```

**Review by brianpursley — CHANGES_REQUESTED** [review]
Left a comment on the issue (see above).  I think instead of removing this line the function should actually be refactored, or at least consider it.

I think the close has to happen before the launch, but there is a better way to do that instead of having to call it twice.

**Pull request closed without merging**